selfdrive/ui/qt/offroad/wifiManager.cc

```diff
@@ -341,9 +341,14 @@ void WifiManager::connectionRemoved(const QDBusObjectPath &path) {
 }
 
 void WifiManager::newConnection(const QDBusObjectPath &path) {
-  knownConnections[path] = getConnectionSettings(path).value("802-11-wireless").value("ssid").toString();
-  if (knownConnections[path] != tethering_ssid) {
-    activateWifiConnection(knownConnections[path]);
+  const Connection &settings = getConnectionSettings(path);
+  if (settings.value("connection").value("type") == "802-11-wireless") {
+    knownConnections[path] = settings.value("802-11-wireless").value("ssid").toString();
+    if (knownConnections[path] != tethering_ssid) {
+      activateWifiConnection(knownConnections[path]);
+    }
+  } else if (path.path() != "/") {
```

Comment:
Should we just delete this 2nd branch then since it's already picked up?

Reply:
Even though I couldn't trigger the second bug/fix with a late LTE startup, technically nothing stops a user from SSHing in and deleting and recreating the LTE connection, and *that* would surely do it. Won't be common, but shouldn't make UI explode either.

For various reasons I went with the simplest possible approach: make sure new-connection import works just like init-time/existing-connection import. That way it doesn't matter when a connection shows up, it's always handled the same way. Perhaps that code should move to a shared function.

Reply:
I meant that we keep the check, but don't update `lteConnectionPath`.

Reply:
I can take it out if you want, but it IS possible (admittedly vanishingly rare) for the LTE connection to be destroyed and/or created at runtime. If you want to keep things simple, I'd rather push the init-time connection handling out into a separate function:

https://github.com/commaai/openpilot/blob/ec71aa0458d7f655be1204e1fa7f3060c5d264d3/selfdrive/ui/qt/offroad/wifiManager.cc#L377-L382

And call that from both `initConnections()` and `newConnection()`, which is probably what I should have done in the first place. Then we can come back and make sure USB Ethernet devices don't break things no matter when they're plugged in, etc. Is that amenable to you?

Reply:
Think it's fine to not deal with LTE being recreated at runtime. That should make things easier to reason about.  It's only used for the roaming toggle anyway.